packages/core/src/browser/shell/application-shell.ts

```diff
@@ -752,24 +753,40 @@ export class ApplicationShell extends Widget {
         let widget = find(this.mainPanel.widgets(), w => w.id === id);
         if (widget) {
             this.mainPanel.activateWidget(widget);
-            return widget;
+            return this.checkActivation(widget);
         }
         widget = find(this.bottomPanel.widgets(), w => w.id === id);
         if (widget) {
             this.expandBottomPanel();
             this.bottomPanel.activateWidget(widget);
-            return widget;
+            return this.checkActivation(widget);
         }
         widget = this.leftPanelHandler.activate(id);
         if (widget) {
-            return widget;
+            return this.checkActivation(widget);
         }
         widget = this.rightPanelHandler.activate(id);
         if (widget) {
-            return widget;
+            return this.checkActivation(widget);
         }
     }
 
+    /**
+     * Focus is taken by a widget through the `onActivateRequest` method. It is up to the
+     * widget implementation which DOM element will get the focus. The default implementation
+     * of Widget does not take any focus. This method can help finding such problems by logging
+     * a warning in case a widget was explicitly activated, but did not trigger a change of the
+     * `activeWidget` property.
+     */
+    private checkActivation(widget: Widget): Widget {
+        window.requestAnimationFrame(() => {
+            if (this.activeWidget !== widget) {
+                this.logger.warn("Widget was activated, but did not accept focus: " + widget.id);
```

Comment:
I have tried it out (removed the if statement to always log); it worked. Unfortunately, this logging does not show up in the browser console but only in the backend console. I do not know whether it was on purpose, but if you would use `console.warn` instead of `logger.warn`, it would log to both places. What do you think?

<img width="751" alt="screen shot 2018-03-30 at 09 40 57" src="https://user-images.githubusercontent.com/1405703/38129347-fb0aba1e-33fe-11e8-93ec-9de48940fa66.png">

```
[2018-03-30T07:40:52.944Z]  WARN: Theia/11837 on prprpr.fritz.box: Widget was activated, but did not accept focus: code-editor-opener:file:///Users/akos.kitta/Desktop/theia/.npmignore []
```

Reply:
Does Electron support the browser console?

Reply:
I would say; yes. (I have not tried it though.)

Reply:
Do we have any guideline on when to use `console` and when to use `ILogger`?

Reply:
I am not aware of any.

Reply:
I changed this to `console.warn`.

Reply:
`console.warn` is good, if you use `ILogger` it will end up only on the backend, via `console` you can log in the browser console and to the backend at the same time.